app/models/apple/config.rb

```diff
@@ -24,27 +20,9 @@ class Config < ApplicationRecord
     delegate :public_feed, to: :podcast, allow_nil: true
     alias_method :private_feed, :feed
 
-    def self.find_or_build_private_feed(podcast)
-      if (existing = podcast.feeds.find_by(slug: DEFAULT_FEED_SLUG, title: DEFAULT_TITLE))
-        # TODO, handle partitions on apple models via the apple_config
-        # Until then it's not safe to have multiple apple_configs for the same podcast
-        Rails.logger.error("Found existing private feed for #{podcast.title}!")
-        Rails.logger.error("Do you want to continue? (y/N)")
-        raise "Stopping find_or_build_private_feed" if $stdin.gets.chomp.downcase != "y"
-
-        return existing
-      end
-      default_feed = podcast.default_feed
-
-      Feed.new(
-        display_episodes_count: default_feed.display_episodes_count,
-        slug: DEFAULT_FEED_SLUG,
-        title: DEFAULT_TITLE,
-        audio_format: DEFAULT_AUDIO_FORMAT,
-        include_zones: ["billboard", "sonic_id"],
-        tokens: [FeedToken.new(label: DEFAULT_TITLE)],
-        podcast: podcast
-      )
+    def find_or_build_apple_feed(podcast)
+      existing_feed = Feed.where(podcast_id: podcast.id, type: "Feed::AppleSubscription")
```

Comment:
Hrm... was your intention to keep the existing method around for now?  Looks like you changed it from a class method to an instance method.

Also, this is going to return a relation, not a single record.  So the next line could return an array of feeds.  Maybe use the subclass directly, plus a rails magic finder?

```ruby
Feed::AppleSubscription.find_by_podcast_id(podcast.id)
```